solidity/dashboard/src/__tests__/saga/coverage-pool.test.js

```diff
@@ -257,104 +268,899 @@ describe("Coverage pool saga test", () => {
-  //       initialCovTotalSupply,
-  //       transferEventData.value
-  //     ).toString()
-  //     const updatedShareOfPool = 0.8
-  //     const estimatedKeepBalance = KEEP.fromTokenUnit(350).toString()
-  //     const estimatedRewards = KEEP.fromTokenUnit(35).toString()
-  //     const updatedTvl = KEEP.fromTokenUnit(10000).toString()
-  //     const keepInUSD = new BigNumber(0.25)
-  //     const updatedAPY = 0.5
-  //     const tvlInUSD = new BigNumber(keepInUSD)
-  //       .multipliedBy(KEEP.toTokenUnit(updatedTvl))
-  //       .toFormat(2)
-
-  //     return expectSaga(subscribeToCovTokenTransferEvent)
-  //       .withReducer(coveragePoolReducer, initialState.coveragePool)
-  //       .withState(initialState)
-  //       .provide([
-  //         [select(selectors.getCoveragePool), initialState],
-  //         [select(selectors.getUserAddress), address],
-  //         [
-  //           matchers.call.fn(Keep.coveragePoolV1.shareOfPool),
-  //           updatedShareOfPool,
-  //         ],
-  //         [
-  //           matchers.call.fn(
-  //             Keep.coveragePoolV1.estimatedCollateralTokenBalance
-  //           ),
-  //           estimatedKeepBalance,
-  //         ],
-  //         [
-  //           matchers.call.fn(Keep.coveragePoolV1.estimatedRewards),
-  //           estimatedRewards,
-  //         ],
-  //         [matchers.call.fn(Keep.coveragePoolV1.totalValueLocked), updatedTvl],
-  //         [
-  //           matchers.call.fn(
-  //             Keep.coveragePoolV1.exchangeService.getKeepTokenPriceInUSD
-  //           ),
-  //           keepInUSD,
-  //         ],
-  //         [matchers.call.fn(Keep.coveragePoolV1.apy), updatedAPY],
-  //       ])
-  //       .dispatch({
-  //         type: COVERAGE_POOL_COV_TOKEN_TRANSFER_EVENT_EMITTED,
-  //         payload: { event: mockedEvent },
-  //       })
-  //       .put(
-  //         covTokenUpdated({
-  //           covBalance: updatedCovBalance,
-  //           covTotalSupply: updatedCovTotalSupply,
-  //           shareOfPool: updatedShareOfPool,
-  //           estimatedKeepBalance,
-  //           estimatedRewards,
-  //           totalValueLocked: updatedTvl,
-  //           totalValueLockedInUSD: tvlInUSD,
-  //           apy: updatedAPY,
-  //         })
-  //       )
-  //       .hasFinalState({
-  //         ...initialState,
-  //         covBalance: updatedCovBalance,
-  //         covTotalSupply: updatedCovTotalSupply,
-  //         shareOfPool: updatedShareOfPool,
-  //         estimatedKeepBalance,
-  //         estimatedRewards,
-  //         totalValueLocked: updatedTvl,
-  //         totalValueLockedInUSD: tvlInUSD,
-  //         apy: updatedAPY,
-  //       })
-  //       .run()
-  //   })
-  // })
+
+  describe("Subscribe to Asset Pool's Deposit event", () => {
+    it("should update data correctly if the `Deposit` event has been emitted by current logged user", () => {
+      const address = "0x086813525A7dC7dafFf015Cdf03896Fd276eab60"
```

Comment:
Some values are duplicated in test cases. Maybe we can define them one level higher?

Reply:
8e0cbf567a0e304548f8bd8a0b9d8ec3efd59e6b